api/Catalog.go

```diff
@@ -0,0 +1,89 @@
+package api
+
+import (
+	"encoding/json"
+	"github.com/aditya/ProjectCatalog/models"
+	"github.com/aditya/ProjectCatalog/services"
+	"github.com/gorilla/mux"
+	"net/http"
+)
+
+type CatalogController struct{
+	CatalogService services.ICatalogService
+}
+
+func (h *CatalogController) CreateProduct(w http.ResponseWriter, r *http.Request){
+	w.Header().Set("Content-Type", "application/json")
+	var newProduct models.Product
+	_ = json.NewDecoder(r.Body).Decode(&newProduct)
+	err := h.CatalogService.CreateProduct(newProduct)
+	if err==nil{
+		json.NewEncoder(w).Encode("Product added successfully")
+		return
+	}
+
+	json.NewEncoder(w).Encode("Encountered error while adding product, product already exists")
```

Comment:
not necessary product already exists, there might some error as well, like DB error, etc